src/utils/WeiBig.ts

```diff
@@ -0,0 +1,78 @@
+import Big, { BigSource } from 'big.js';
+
+const numberRegex = /^[0-9]*(\.[0-9]*)?$/;
+
+type RbtcMultiple = 'wei' | 'kwei' | 'mwei' | 'gwei' | 'rbtc';
+
+export default class WeiBig extends Big {
+  constructor(src: number | string | Big, currency: RbtcMultiple) {
+    const safeSrc = src ? src.toString() : '0';
+    const safeBig: Big = src instanceof Big
+      ? src : Big(numberRegex.test(safeSrc) ? safeSrc : '0');
+    switch (currency) {
+      case 'wei':
+        super(safeBig.toFixed(0));
+        break;
+      case 'mwei':
+        super(safeBig.mul(100_000).toFixed(0));
```

Comment:
isn't megawei should be 1000_000?

Reply:
Yes it is, let me fix it and add some tests

Reply:
Will be treated in a new PR.